docs/reference/ingress/kubernetes-with-cert-manager.md

```diff
@@ -0,0 +1,168 @@
+
+Replace `<your-email-here>` with the contact email that will be shown with the SSL certificate.
+
+```yaml
+# letsencrypt-issuer.yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-prod
+  namespace: openfaas
+spec:
+  acme:
+    # Email address used for ACME registration
+    email: <your-email-here>
+    http01: {}
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      key: ""
+      name: letsencrypt-prod
+    server: https://acme-v01.api.letsencrypt.org/directory
+---
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-staging
+  namespace: openfaas
+spec:
+  acme:
+    server: https://acme-staging.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: <your-email-here>
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      name: letsencrypt-staging
+    http01: {}
+```
+
+```sh
+$ kubectl apply -f letsencrypt-issuer.yaml
+```
+
+This will allow `cert-manager` to automatically provision Certificates just in the `openfaas` namespace.
+
+## Install openfaas
+
+The OpenFaaS Helm Chart already supports the nginx-ingress, but we want to customize it further. This is easiest with a custom values file. Below, we enable and configure the ingress object to use our certificate and expose just the gateway
+
+```yaml
+# our-values.yaml
+functionNamespace: openfaas-fn
+ingress:
+    enabled: true
+    annotations:
+        kubernetes.io/ingress.class: nginx
+        certmanager.k8s.io/cluster-issuer: letsencrypt-prod
+    ingress:
+        tls:
+        - hosts:
+          - openfaas.mydomain.com
+          secretName: openfaas-crt
+        hosts:
+        - host: openfaas.mydomain.com
+            serviceName: gateway
+            servicePort: 8080
+            path: /
+```
+
+```sh
+$ helm repo add openfaas https://openfaas.github.io/faas-netes/
+$ helm repo update
+$ helm upgrade openfaas \
+    --install \
+    --namespace openfaas \
+    --values our-values.yaml \
+    openfaas/openfaas
+```
+
+## Create a certificate for your openfaas
+
+Finally, we can create the Certificate resource which triggers the actual creation of the certificate by `cert-manager`
+
+```yaml
+# openfaas-crt.yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: openfaas-crt
+spec:
+  secretName: openfaas-crt
+  dnsNames:
+    - openfaas.mydomain.com
+  acme:
+    config:
+      - http01:
+          ingressClass: nginx
+        domains:
+          - openfaas.mydomain.com
+  issuerRef:
+    name: letsencrypt-prod
+    kind: ClusterIssuer
+```
+
+```sh
+$ kubectl apply -f openfaas-crt.yaml
+```
+
```

Comment:
Perhaps include a troubleshooting section with the commands we are used to using to check for the cert / issuer and nginx logs.